### PR TITLE
add node-manager chart

### PR DESCRIPTION
**Pull request by chrisho**
- Initialize Node Manager code.
- Add Ksmtuned Controller manage KSM of node.

**Related issue**
https://github.com/harvester/harvester/issues/2302

**Dependency PR**
https://github.com/harvester/node-manager/pull/1

### CODE REVIEW

File: charts/harvester-node-manager/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apiVersion: v2
+name: harvester-node-manager
+description: A Helm chart for Harvester Node Disk Manager
```
Comment (guangbochen) [inline charts/harvester-node-manager/Chart.yaml:3]:
```suggestion
description: A Helm chart for Harvester Node Manager
```

File: charts/harvester-node-manager/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "harvester-node-manager.name" . }}
+rules:
+  - apiGroups: [ "apiextensions.k8s.io" ]
+    resources: [ "customresourcedefinitions" ]
+    verbs: [ "*" ]
```
Comment (guangbochen) [inline charts/harvester-node-manager/templates/rbac.yaml:16]: why does it need full permission of the CRD resource?

File: charts/harvester-node-manager/values.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Default values for harvester-node-manager.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+image:
+  repository: rancher/harvester-node-manager
+  pullPolicy: IfNotPresent
```
Comment (guangbochen) [inline charts/harvester-node-manager/values.yaml:6]: /s/IfNotPresent/Always

File: charts/harvester-node-manager/values.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "master-head"
+
+nameOverride: ""
+
+# Specify whether to enable VIP, defaults to false
+vipEnabled: false
```
Comment (guangbochen) [inline charts/harvester-node-manager/values.yaml:13]: why does it need `vipEnabled`?

File: charts/harvester-node-manager/templates/daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          env:
+            - name: NODENAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
+            - name: HOST_PROC
+              value: /host/proc
+          securityContext:
+            privileged: true
```
Comment (bk201) [inline charts/harvester-node-manager/templates/daemonset.yaml:38]: Is there a way to not run a privileged container? Perhaps just adding required caps.
Doing a search and seems `procMount`  could be helpful, but it doesn't cover sysfs.

Comment (bk201) [reply charts/harvester-node-manager/templates/daemonset.yaml:38]: We can use privileged until we find a better solution. Thanks!

**Review by bk201 — APPROVED** [review]

**Comment by chrisho** [comment]
Image repo and tags are used temporarily.

File: charts/harvester-node-manager/values.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Default values for harvester-node-manager.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+image:
+  repository: ghcr.io/harvester/node-manager
+  pullPolicy: Always
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "v0.1.0"
```
Comment (guangbochen) [inline charts/harvester-node-manager/values.yaml:8]: should be `latest` as this is a master branch.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

File: charts/harvester-node-manager/templates/daemonset.yaml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    {{- end }}
+      labels:
+        {{- include "harvester-node-manager.selectorLabels" . | nindent 8 }}
+        name: harvester-node-manager
+    spec:
+      serviceAccountName: {{ include "harvester-node-manager.name" . }}
+      containers:
+        - name: node-manager
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: IfNotPresent
```
Comment (futuretea) [inline charts/harvester-node-manager/templates/daemonset.yaml:26]: can not be overwrited by values.yaml

**Review by futuretea — APPROVED** [review]
LGTM

**Pull request merged**